docs: initial Diataxis site (#248)
jnsgruk/concierge: docs/reference/configuration.md

---
myst:
  html_meta:

>     description: The complete YAML schema for Concierge configuration files, covering Juju, providers (LXD, K8s, MicroK8s, Google), and host packages and snaps.

Reviewer: ```suggestion
    description: The complete YAML schema for Concierge configuration files, covering Juju, providers (LXD, Canonical Kubernetes, MicroK8s, Google), and host packages and snaps.
```

---

(reference-configuration)=
# Configuration schema

Concierge reads a YAML configuration file. The schema below describes every field. All top-level blocks are optional; a minimal configuration might set just one provider.

> For task-oriented guidance, see [Write a custom config](../how-to/write-a-custom-config).

Reviewer: ```suggestion
For task-oriented guidance, see {ref}`how-to-write-a-custom-config`.
```

  agent-version: <version>
  # (Optional) A map of model-defaults to set when bootstrapping *all* Juju
  # controllers.
  model-defaults:
    <model-default>: <value>
  # (Optional) A map of bootstrap-constraints to set when bootstrapping *all*
  # Juju controllers.
  bootstrap-constraints:
    <bootstrap-constraint>: <value>
  # (Optional) Extra arguments to append to the `juju bootstrap` command.
  # Parsed using shell-style splitting rules.
  extra-bootstrap-args: <args>

# (Required) Providers to install and bootstrap.
providers:
  microk8s:
    enable: true | false
    bootstrap: true | false
    channel: <channel>
    model-defaults:
      <model-default>: <value>
    bootstrap-constraints:
      <bootstrap-constraint>: <value>
    # (Optional) MicroK8s addons to enable.
    addons:
      - <addon>[:<params>]
    # (Optional) Image registry mirror. Values support ${VAR} interpolation.
    image-registry:
      url: <url>
      username: <username>
      password: <password>

  k8s:
    enable: true | false
    bootstrap: true | false
    channel: <channel>
    model-defaults:
      <model-default>: <value>
    bootstrap-constraints:
      <bootstrap-constraint>: <value>

>     # (Optional) K8s features to configure.

Reviewer: ```suggestion
    # (Optional) Canonical Kubernetes features to configure.
```

---
myst:
  html_meta:
    description: The complete YAML schema for Concierge configuration files, covering Juju, providers (LXD, K8s, MicroK8s, Google), and host packages and snaps.
---

(reference-configuration)=
# Configuration schema

Concierge reads a YAML configuration file. The schema below describes every field. All top-level blocks are optional; a minimal configuration might set just one provider.

For task-oriented guidance, see [Write a custom config](../how-to/write-a-custom-config).

```yaml
# (Optional) Target Juju configuration.
juju:
  # (Optional) Disable installation of Juju (and therefore all bootstrapping).
  disable: true | false
  # (Optional) Channel from which to install Juju.
  channel: <channel>

>   # (Optional) Snap revision from which to install Juju, for example "31429". When

Reviewer: Make the comment slightly shorter. Same for the 3.6.11 comment
